File: src/main/java/org/leetcode/climbingStairs_70.java
package org.leetcode;

import java.util.HashMap;

public class climbingStairs_70 {
    public static HashMap<Integer, Integer> map = new HashMap<>();
    public int climbStairs(int n) {
        if(n == 1)
            return 1;
        else if(n == 2)
            return 2;
        if(map.containsKey(n))
            return map.get(n);
        int result = climbStairs(n-1) + climbStairs(n-2);
        map.put(n, result);

        return result;
    }
}

File: src/main/python/ClimbingStairs_70.py
class Solution:
    dp = {}
    def climbStairs(self, n: int) -> int:
        if n in self.dp:
            return self.dp[n]
        if n == 1:
            return 1
        if n == 2:
            return 2
        self.dp[n] = self.climbStairs(n - 2) + self.climbStairs(n - 1)
        return self.dp[n]